tests/char.rs

```diff
@@ -0,0 +1,23 @@
+use serde_bin_prot::to_writer;
+
+mod common;
+use common::TestCase;
+
+fn char_test_cases() -> Vec<TestCase<char>> {
+    vec![
+        TestCase::new('\u{000}', vec![0x00]),
+        TestCase::new('A', vec![0x41]),
+        TestCase::new('z', vec![0x7a]),
+        TestCase::new(';', vec![0x3b]),
+        // TestCase::new('\u{255}', vec![0xff]),
```

Comment:
I guess this means OCaml chars are a single byte where Rust chars are utf-8. To comply with the spec we will have to change the serialize char implementation and error on any chars outside the single byte range